pkg/closenicely/closeutil.go

```diff
@@ -0,0 +1,16 @@
+package closenicely
+
+import (
+	"go.uber.org/zap"
+	"io"
+)
+
+func OrDebug(closer io.Closer) {
+	FuncOrDebug(closer.Close)
+}
+
+func FuncOrDebug(closer func() error) {
+	if err := closer(); err != nil {
+		zap.L().Debug("Failed to close resource", zap.Error(err))
+	}
+}
```

Comment:
Something I've used on side projects which might be nice (adapted to the style of this package):
```go
func OrDebug(o any) {
	var err error
	switch o := o.(type) {
	case io.Closer:
		err = o.Close()
	case func() error:
		err = o()
	default:
		return
	}
	if err != nil {
		zap.L().Debug("Failed to close resource", zap.Error(err))
	}
}
```